Source/ablastr/utils/IntervalsParser.cpp

```diff
@@ -0,0 +1,176 @@
+/* Copyright 2022 The ABLASTR Community
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ * Authors: Neil Zaim, Luca Fedeli, Axel Huebl
+ */
+#include "ablastr/utils/IntervalsParser.H"
+
+#include "ablastr/utils/text/StringUtils.H"
+//include "ParserUtils.H"
+#include "Utils/TextMsg.H"
+
+#include <AMReX_Utility.H>
+
+#include <algorithm>
+
+
+ablastr::utils::parser::SliceParser::SliceParser (
+    const std::string& instr,
+    const bool isBTD
+):
+    m_isBTD{isBTD}
+{
+    // split string and trim whitespaces
+    auto insplit = ablastr::utils::text::split_string<std::vector<std::string>>(
+        instr, m_separator, true);
+
+    if(insplit.size() == 1){ // no colon in input string. The input is the period.
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(!m_isBTD, "must specify interval stop for BTD");
+        m_period = parseStringtoInt(insplit[0], "interval period");}
+    else if(insplit.size() == 2) // 1 colon in input string. The input is start:stop
+    {
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(!m_isBTD || !insplit[1].empty(), "must specify interval stop for BTD");
+        if (!insplit[0].empty()){
+            m_start = parseStringtoInt(insplit[0], "interval start");}
```

Comment:
`parseStringtoInt`: @WeiqunZhang can I use an AMReX intrinsic for this now?

Reply:
I looked at the implementation of that function. It use amrex::Parser to parse it as double precision numbers and then check if it can be safely cast to int. There is amrex::IParser we can use. It parses the numbers as long long. So the chance of overflowing during parsing is tiny. The safety checking for casting to int would also be much simpler.